File: streamx-console/streamx-console-service/src/main/java/com/streamxhub/streamx/console/system/entity/Team.java
/*
 * Copyright (c) 2019 The StreamX Project
 *
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.streamxhub.streamx.console.system.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.extern.slf4j.Slf4j;

import java.util.Date;
import java.util.List;

/**
 * @author daixinyu
 */
@Data
@TableName("t_team")
@Slf4j
public class Team {

    @TableId(value = "TEAM_ID", type = IdType.AUTO)
    private Long teamId;

    private String teamCode;

    @TableId(value = "TEAM_NAME")
    private String teamName;

    private Date createTime;

    // 排序字段
    private transient String sortField;

    // 排序规则 ascend 升序 descend 降序
    private transient String sortOrder;

    private transient String createTimeFrom;
    private transient String createTimeTo;
    private transient List<Long> teamIdList;
}

File: streampark-console/streampark-console-webapp/src/api/team.js
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import api from './index'
import http from '@/utils/request'

export function post (queryParam) {
  return http.post(api.Team.POST, queryParam)
}

export function update (queryParam) {
  return http.put(api.Team.UPDATE, queryParam)
}

export function list (queryParam) {
  return http.post(api.Team.LIST, queryParam)
}

export function users (queryParam) {
  return http.post(api.Team.USERS, queryParam)
}

export function checkName (queryParam) {
  return http.post(api.Team.CHECK_NAME, queryParam)
}

export function remove (queryParam) {
  return http.delete(api.Team.DELETE, queryParam)
}
